### PR TITLE
Bump go version to go 1.17.8

### PR DESCRIPTION
**Pull request by umbynos**
**Please check if the PR fulfills these requirements**

- [X] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [X] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**

Update go version used
- **What is the current behavior?**

We currently use go 1.16.4
* **What is the new behavior?**

We currently use go 1.17.8
- **Does this PR introduce a breaking change, and is
[titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**

nope
* **Other information**:

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

**Comment by umbynos** [comment]
I have run the tests locally and it seems there are no regressions with go 1.17

**Comment by rsora** [comment]
Let's be cautious on this one, I'll wait for other tasks to be closed, and also to dig a bit more with testing :mag_right:

**Comment by umbynos** [comment]
BTW @silvanocerza finished running local tests on macOS and everything went smooth

**Comment by umbynos** [comment]
Note for future self: The homebrew  version is built with go 1.17 (and not pinned) since 0.20.0 https://github.com/Homebrew/homebrew-core/pull/89858

**Review by silvanocerza — APPROVED** [review]

**Comment by silvanocerza** [comment]
@rsora @umbynos I think we can merge this, I have been building the `arduino-cli` with Go 1.17 since it came out and never had any issue connected to it.

File: DistTasks.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ version: "3"
 
 vars:
   CONTAINER: "docker.elastic.co/beats-dev/golang-crossbuild"
-  GO_VERSION: "1.16.4"
+  GO_VERSION: "1.17.3"
```
Comment (per1234) [inline DistTasks.yml:22]:
```suggestion
  GO_VERSION: "1.17.8"
```
This is the current latest available:
https://github.com/elastic/golang-crossbuild/releases/tag/1.17.8

Unless there is a specific reason for using 1.17.3, I suggest we get all the way up to date while we are at it.

**Review by per1234 — CHANGES_REQUESTED** [review]

**Review by per1234 — CHANGES_REQUESTED** [review]
These should also be updated to say 1.17:

- https://github.com/arduino/arduino-cli/blob/master/docs/CONTRIBUTING.md#prerequisites
- https://github.com/arduino/arduino-cli/blob/master/docs/CONTRIBUTING.md#local-development

(they were forgotten during the Go 1.14 -> 1.16 transition)

**Review by per1234 — APPROVED** [review]

**Pull request merged**